engine/src/test/java/io/zeebe/engine/processor/workflow/incident/MappingIncidentTest.java

```diff
@@ -0,0 +1,582 @@
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .getFirst();
+
+    ENGINE.variables().ofScope(failureEvent.getKey()).withDocument(VARIABLES).update();
+    ENGINE.incident().ofInstance(workflowInstanceKey).resolve();
+
+    // create a second incident
+    workflowInstanceKey = ENGINE.workflowInstance().ofBpmnProcessId("process").create();
+    failureEvent =
+        RecordingExporter.workflowInstanceRecords()
+            .withElementId("failingTask")
+            .withIntent(WorkflowInstanceIntent.ELEMENT_ACTIVATING)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .getFirst();
+
+    final long secondIncidentKey =
+        RecordingExporter.incidentRecords(CREATED)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .getFirst()
+            .getKey();
+
+    // when
+    ENGINE.variables().ofScope(failureEvent.getKey()).withDocument(VARIABLES).update();
+    final Record incidentResolvedEvent =
+        ENGINE.incident().ofInstance(workflowInstanceKey).resolve();
+
+    // then
+    assertThat(incidentResolvedEvent.getKey()).isEqualTo(secondIncidentKey);
+  }
+
+  @Test
+  public void shouldResolveIncidentIfActivityTerminated() {
+    // given
+    ENGINE.deployment().withXmlResource(WORKFLOW_INPUT_MAPPING).deploy();
+
+    final long workflowInstanceKey = ENGINE.workflowInstance().ofBpmnProcessId("process").create();
+
+    final Record incidentCreatedEvent =
+        RecordingExporter.incidentRecords()
+            .withIntent(CREATED)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .getFirst();
+
+    // when
+    ENGINE.workflowInstance().withInstanceKey(workflowInstanceKey).cancel();
+
+    // then
+    final Record activityTerminated =
+        RecordingExporter.workflowInstanceRecords()
+            .withElementId("failingTask")
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .withIntent(WorkflowInstanceIntent.ELEMENT_TERMINATED)
+            .getFirst();
+
+    final Record<IncidentRecordValue> incidentResolvedEvent =
+        RecordingExporter.incidentRecords()
+            .withIntent(RESOLVED)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .getFirst();
+
+    assertThat(incidentResolvedEvent.getKey()).isEqualTo(incidentCreatedEvent.getKey());
+    assertThat(activityTerminated.getPosition())
+        .isEqualTo(incidentResolvedEvent.getSourceRecordPosition());
+
+    Assertions.assertThat(incidentResolvedEvent.getValue())
+        .hasErrorType(ErrorType.IO_MAPPING_ERROR.name())
+        .hasErrorMessage("No data found for query foo.")
+        .hasBpmnProcessId("process")
+        .hasWorkflowInstanceKey(workflowInstanceKey)
+        .hasElementId("failingTask")
+        .hasElementInstanceKey(incidentResolvedEvent.getValue().getElementInstanceKey())
+        .hasVariableScopeKey(incidentResolvedEvent.getValue().getElementInstanceKey());
+  }
+
+  @Test
+  @Category(UnstableTest.class)
```

Comment:
Is this still necessary? I'd assume that the previous flakiness is gone, I also couldn't get it to fail locally

Reply:
I think you're right we can remove it nice catch